dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessInstanceMapper.xml

```diff
@@ -44,7 +44,7 @@
         <foreach collection="states" item="i" open="(" close=")" separator=",">
             #{i}
         </foreach>
-        order by id asc
+        order by process_instance_priority, id asc
```

Comment:
I'm not quite sure if this is nessnary to do this.

I think we can provide users with a sort item, which is similar to excel and gives users free choice.

WDYT? @zhuxt2015 @zhongjiajie @caishunfeng

Reply:
I think sort by process_instance_priority is most suitable and most commonly used . why do you  want to add sort item, Can you give some cases?

Reply:
I think we don't need to change this part, we need to set the processInstance's priority in to failover command is enough. if you query by priority here, you still need to set the priority of command.

Reply:
Sort item like this.

![](https://vip2.loli.io/2022/07/04/iIoc9auHqyb5gpd.png)